src/js/logger.js

```diff
@@ -0,0 +1,101 @@
+"use strict";
+var fluid = require("infusion");
+
+require("json5/lib/register");
+
+fluid.registerNamespace("fluid.lintAll.logger");
+
+fluid.lintAll.logger.sortByPosition = function (a, b) {
+    if (a.line && b.line) {
+        if (a.line === b.line) {
+            if (a.column && b.column) {
```

Comment:
Feels like both of these compound checks could be rolled up into something like

      return (a.column || 0) - (b.column || 0);

Although to be honest the logical operator is just a courtesy since numeric coercion would do the rest - try evaluating 'null - 3' in the console : P

Reply:
very useful pattern, thanks.